src/main/java/nextstep/ladder/domain/Line.java

```diff
@@ -24,57 +26,64 @@ public int getLadderLength() {
     }
 
     public List<Boolean> getLine() {
-        List<Boolean> deepCopiedPoints = new ArrayList<>();
-        deepCopiedPoints.addAll(points);
-        return deepCopiedPoints;
+        return points.stream()
+            .map(Point::hasPoint)
+            .collect(Collectors.toList());
     }
 
-    private List<Boolean> makePoints(int length, GenerableStrategy generableStrategy) {
-        List<Boolean> points = initializePoints(length);
-
-        for (int current = 0; current < length; current++) {
-            Boolean hasPoint = generableStrategy.generate();
-            if (hasPoint && canBePoint(points, current)) {
-                points.set(current, true);
-            }
-        }
-        return points;
-    }
-
-    private List<Boolean> initializePoints(int length) {
-        List<Boolean> points = new ArrayList<>();
-        for (int i = 0; i < length; ++i) {
-            points.add(false);
+    private List<Point> makePoints(int length, GenerableStrategy generableStrategy) {
+        points = new ArrayList<>();
+        for (int currentPosition = 0; currentPosition < length; currentPosition++) {
+            points.add(Point.makePointInLine(currentPosition, this, generableStrategy));
         }
         return points;
     }
 
-    private boolean canBePoint(List<Boolean> points, int currentPosition) {
-        if (hasPreviousPositionPoint(points, currentPosition)) {
-            return false;
-        }
-        if (hasNextPositionPoint(points, currentPosition)) {
-            return false;
+    private boolean isInLine(int position) {
+        if (position >= 0 && position <= points.size() - 1) {
+            return true;
         }
-        return true;
+        return false;
     }
 
-    private boolean hasPreviousPositionPoint(List<Boolean> points, int currentPosition) {
+    public boolean hasPreviousPositionPoint(int currentPosition) {
         int previousPosition = currentPosition - 1;
-        if (previousPosition >= 0 && points.get(previousPosition)) {
+        if (isInLine(previousPosition) && points.get(previousPosition).hasPoint()) {
             return true;
         }
         return false;
```

Comment:
다른 대상에 대한 검사를 하나의 if 문 안에서 비교하고 있어요.
아래 처럼 하나의 대상에 대해서만 비교하도록 변경해 보는건 어떨까요?
```suggestion
        if (!isInLine(previousPosition)) {
            return false;
        }
    
        Point previousPoint = points.get(previousPosition);
        return previousPoint.hasPoint();
```